controller/src/test/java/org/jboss/as/controller/persistence/ConfigurationFileUnitTestCase.java

```diff
@@ -0,0 +1,135 @@
+package org.jboss.as.controller.persistence;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+
+/**
+ *
+ * @author <a href="padamec@redhat.com">Petr Adamec</a>
+ */
+public class ConfigurationFileUnitTestCase {
+    private static final String UNCORRECT_NAME = "standalone.xml";
+    private static final String CORRECT_NAME = "20180101-000000000standalone.xml";
+    private static final String CONFIGURATION_DIR = "WFCORE-3513";
+    private static final String HISTORY_DIRECTORY = "standalone_xml_history";
+    private static final String SNAPSHOT_DIRECTORY = "snapshot";
+
+    File configurationDir;
+    File configurationFile;
+    File historyDirectory;
+    File snapshotsDirectory;
+
+    @Before
+    public void createDirectoryAndFile() throws Exception {
+        createConfigurationDir();
+        historyDirectory = new File(configurationDir, HISTORY_DIRECTORY);
+        historyDirectory.mkdir();
+        if(!historyDirectory.exists()){
+            throw new RuntimeException("cannot create dir "+HISTORY_DIRECTORY);
+        }
+        snapshotsDirectory = new File(historyDirectory, SNAPSHOT_DIRECTORY);
+        snapshotsDirectory.mkdir();
+        if(!snapshotsDirectory.exists()){
+            throw new RuntimeException("cannot create dir "+SNAPSHOT_DIRECTORY);
+        }
+    }
+
+    /**
+     * Create configuration directory for this test at tmp directory
+     */
+    private void createConfigurationDir(){
+        File tmpRoot = new File(System.getProperty("java.io.tmpdir"));
+        long currentTimeMillis = System.currentTimeMillis();
+        configurationDir = new File(tmpRoot, CONFIGURATION_DIR);
+        for (int i = 0; i < 200; i++) {
+            configurationDir = new File(tmpRoot, String.valueOf(currentTimeMillis++));
+            if (!configurationDir.exists() && configurationDir.mkdirs()) {
+                break;
+            }
+        }
+
+        if (!configurationDir.exists()) {
+            throw new RuntimeException("cannot create configuration dir");
+        }
+    }
+
+    /**
+     * Create file with content
+     * @param dir parent directory
+     * @param name name of file
+     * @param contents content of file
+     * @return new File with name and contents at the parent directory
+     * @throws IOException
+     */
+    private File createFile(File dir, String name, String contents) throws IOException {
+        File newFile = new File(dir, name);
```

Comment:
First check if parent exists, if not - fail. You could refactor the parent check into a separate method and use it here and in createDirectory